tests/test_dcrModel.py

```diff
@@ -241,13 +267,70 @@ def testDcrCalculation(self):
         wcs = self.makeDummyWcs(rotAngle, pixelScale, crval=visitInfo.getBoresightRaDec())
         dcrShift = calculateDcr(visitInfo, wcs, filterInfo, dcrNumSubfilters)
         # Compare to precomputed values.
-        refShift = [(-0.5363512808, -0.3103517169),
-                    (0.001887293861, 0.001092054612),
-                    (0.3886592703, 0.2248919247)]
+        refShift = [(-0.5575046818186551, -0.2705169391233882),
+                    (0.001961727697724214, 0.0009518854091965191),
+                    (0.40398777927558477, 0.1960262237374977)]
         for shiftOld, shiftNew in zip(refShift, dcrShift):
             self.assertFloatsAlmostEqual(shiftOld[1], shiftNew[1], rtol=1e-6, atol=1e-8)
             self.assertFloatsAlmostEqual(shiftOld[0], shiftNew[0], rtol=1e-6, atol=1e-8)
 
+    def testCoordinateTransformDcrCalculation(self):
+        """Check the DCR calculation using astropy coordinate transformations.
+
+        Astmospheric refraction causes sources to appear closer to zenith than
+        they really are. An alternate calculation of the shift due to DCR is to
+        transform the pixel coordinates to altitude and azimuth, add the DCR
+        amplitude to the altitude, and transform back to pixel coordinates.
+        """
+        dcrNumSubfilters = 3
+        afwImageUtils.defineFilter("gTest", self.lambdaEff,
+                                   lambdaMin=self.lambdaMin, lambdaMax=self.lambdaMax)
+        filterInfo = afwImage.Filter("gTest")
+        lambdaEff = filterInfo.getFilterProperty().getLambdaEff()
+        pixelScale = 0.2*arcseconds
+
+        for testIter in range(self.nRandIter):
+            rotAngle = 360.*self.rng.rand()*degrees
+            azimuth = 360.*self.rng.rand()*degrees
+            elevation = (45. + self.rng.rand()*40.)*degrees  # Restrict to 45 < elevation < 85 degrees
+            visitInfo = self.makeDummyVisitInfo(azimuth, elevation)
+            loc = EarthLocation(lat=visitInfo.getObservatory().getLatitude().asDegrees()*u.degree,
```

Comment:
The location is the same for each iteration, but it is calculated from `visitInfo` which is only available inside the loop. I could make the observatory coordinates globals, or construct a dummy `visitInfo` before the loop, but this felt cleaner. I'll add a comment.

Reply:
Or ask for a "real" VisitInfo outside the loop and calculate the `loc` once...

Reply:
I'm sorry, I don't understand what you mean. The visitInfo changes for each iteration since it requires the pointing information, so I don't know what it would mean to have it outside the loop.

Reply:
I mean call it additionally once outside the loop to get a visitInfo you are only going to use to create the location object. Then create new visitInfos within the loop as now.  It's probably not worth doing given that it's not going to appreciably slow down the loop.